src/commands/database-remove.js

```diff
@@ -38,32 +40,199 @@ module.exports = new Command("database:remove <path>")
           ". Are you sure?",
       },
     ]).then(function() {
+      if (!options.concurrent) {
+        options.concurrent = 100;
+      }
       if (!options.confirm) {
         return utils.reject("Command aborted.", { exit: 1 });
       }
-      var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + ".json?";
-      var reqOptions = {
-        url: url,
-        json: true,
-      };
 
-      return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+      function deletePath(path) {
         return new Promise(function(resolve, reject) {
-          request.del(reqOptionsWithToken, function(err, res, body) {
-            if (err) {
+          var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + ".json?";
+          var reqOptions = {
+            url: url,
+            json: true,
+          };
+          return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+            request.del(reqOptionsWithToken, function(err, res, body) {
+              if (err) {
+                return reject(
+                  new FirebaseError("Unexpected error while removing data at " + path, {
+                    exit: 2,
+                  })
+                );
+              } else if (res.statusCode >= 400) {
+                return reject(responseToError(res, body));
+              }
+              if (options.verbose) {
+                utils.logSuccess("Sucessfully removed data at " + path);
+              }
+              return resolve();
+            });
+          });
+        });
+      }
+
+      // return whether it times out or not
+      function prefetchTest(path) {
+        var url = utils.addSubdomain(api.realtimeOrigin, options.instance) + path + ".json?";
+        url += querystring.stringify({
+          timeout: "100ms",
+        });
+        var reqOptions = {
+          url: url,
+        };
+        return api.addRequestHeaders(reqOptions).then(function(reqOptionsWithToken) {
+          return new Promise(function(resolve, reject) {
+            request.get(reqOptionsWithToken, function(err, res, body) {
+              if (err) {
+                return reject(
+                  new FirebaseError("Unexpected error while prefetching data to delete", {
+                    exit: 2,
+                  })
+                );
+              } else if (res.statusCode == 200) {
+                if (res) {
+                  return resolve("small");
```

Comment:
This string enum thing feels a little weird to me. I don't have a clear different solution, but I'm not sure I like it...